src/game/start.py

```diff
@@ -5,133 +5,131 @@
+    """Definition of the user interface and the interactions:"""
 
-    game_state = GameState(parameter)
-    if len(error_message) > 0:
-        game_state.message = error_message
+    def start_game(self):
+        print("Welcome to Quarto-Py")
 
-    if (len(game_state.message) > 0):
-        game_state.message += """\nValid sample : --state='{"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}'"""
+        initial_state, error_parsing = self.get_state_parameter(sys.argv)
+        pieces_list = Piece.create_pieces_list()
 
-    while len(game_state.remaining_pieces) > 0:
-        display_game(game_state)
-        prompt_piece_selection(game_state)
-        game_state.swich_player()
-        display_game(game_state)
-        prompt_piece_location(game_state)
+        game_state = GameState()
 
+        if len(error_parsing) > 0:
+            game_state.message = error_parsing
+        elif type(initial_state) is dict:
+            game_state.load_state(initial_state)
 
-def prompt_piece_selection(game_state):
-    value_ok = False
-    while not value_ok:
-        piece = input("Choose the next piece of the opponent : ")
-        try:
-            piece = int(piece)
-            valid_piece = game_state.check_piece_validity(piece)
-            if not valid_piece:
+        if (len(game_state.message) > 0):
+            game_state.message += """\nValid sample : --state='{"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}'"""
+
+        while len(game_state.remaining_pieces) > 0:
+            self.display_game(game_state)
+            self.prompt_piece_selection(game_state)
+            game_state.swich_player()
+            self.display_game(game_state)
+            self.prompt_piece_location(game_state)
+
+    def prompt_piece_selection(self, game_state):
+        while True:
+            try:
+                piece = int(input("Choose the next piece of the opponent : "))
+                valid_piece = game_state.check_piece_availability(piece)
+                if valid_piece:
+                    break
                 game_state.message = "You must choose a number available in the list"
-                display_game(game_state)
+                self.display_game(game_state)
+
+            except ValueError:
+                game_state.message = "You have to type number between 1 and " + str(PIECES_NUMBER)
+                self.display_game(game_state)
+            except KeyboardInterrupt:
+                print("\nGame aborted")
+                exit()
+        game_state.game_turn.selected_piece = piece
+
+    def prompt_piece_location(self, game_state):
+        position = input("Choose the position to place your piece : ")
+        # basic remove of the piece to end correctly the "while", but will be replace by real placement for next story
+        game_state.remaining_pieces.remove(game_state.game_turn.selected_piece)
+
+    def get_state_parameter(self, argv):
```

Comment:
The function name does not reflect that argv is used as input here.
I'll name this function "parse_state_from_args" for example